app/models/organization.rb

```diff
@@ -392,6 +392,28 @@ def self.upload_invoice(file_path,file_name)
     end
   end
 
+
+  def self.upload_commission_statement(file_path,file_name)
+    statement_date = invoice_date(file_path) rescue nil
+    org = by_invoice_filename(file_path) rescue nil
```

Comment:
I wonder if it would be better to create a generic parsers for invoice, commission, etc. Will they also have the same format as hbx_id_MMDDYYYY format??

Reply:
Since I think all of these methods will soon be deprecated by the transport gateway instead of building out a fully generic system it is probably better and more efficient to have a unique parser right now. The parser for invoices does need to be made more granular and we should open a ticket to addess that. As such, it will return false if any organization has a document that is being attempted to be uploaded on the same date. The custom commission parser avoids this collision.